Tasks/GooglePlayReleaseV4/GooglePlay.ts

```diff
@@ -0,0 +1,489 @@
+}
+
+/**
+ * Gets the right apk(s) depending on the action. Uses `getApksOrAabs()`
+ * @param action user's action
+ * @returns a list of apks
+ */
+function getApks(action: Action): string[] {
+    return getApksOrAabs(action, 'SingleApk', 'apkFile', 'apkFiles');
+}
+
+/**
+ * Gets the right apk(s)/aab(s) depending on the action.
+ * This function exists to avoid code duplication: the process of getting APKs and AABs is very similar.
+ * @param action user's action
+ * @param singleAction which action would be considered a single file upload
+ * @param singleInput input containing single file pattern. Used if `action == singleAction`
+ * @param multiInput input containing multiple files patterns. Used if `action == 'MultiApkAab'`
+ * @returns a list of apks/aabs
+ */
+function getApksOrAabs(
+    action: Action,
+    singleAction: 'SingleApk' | 'SingleBundle',
+    singleInput: 'apkFile' | 'bundleFile',
+    multiInput: 'apkFiles' | 'bundleFiles',
+): string[] {
+    if (action === singleAction) {
+        const pattern: string = tl.getInput(singleInput, true);
+        const path: string = resolveGlobPath(pattern);
+        tl.checkPath(path, singleInput);
+        return [path];
+    } else if (action === 'MultiApkAab') {
+        const patterns: string[] = tl.getDelimitedInput(multiInput, '\n');
+        const allPaths = new Set<string>();
+        for (const pattern of patterns) {
+            const paths: string[] = resolveGlobPaths(pattern);
+            paths.forEach((path) => allPaths.add(path));
+        }
+        return Array.from(allPaths);
+    }
+
+    return [];
+}
+
+/**
+ * Update a given release track with the given information
+ * Assumes authorized
+ * @param packageName unique android package name (com.android.etc)
+ * @param track one of the values {"internal", "alpha", "beta", "production"}
+ * @param bundleVersionCode version code of uploaded modules.
+ * @param versionCodeFilterType type of version code replacement filter, i.e. 'all', 'list', or 'expression'
+ * @param versionCodeFilter version code filter, i.e. either a list of version code or a regular expression string.
+ * @param userFraction the fraction of users to get update
+ * @param updatePriority - In-app update priority value of the release. All newly added APKs in the release will be considered at this priority. Can take values in the range [0, 5], with 5 the highest priority. Defaults to 0.
+ * @returns track A promise that will return result from updating a track
+ *                            { track: string, versionCodes: [integer], userFraction: double }
+ */
+async function updateTrack(
+    edits: pub3.Resource$Edits,
+    packageName: string,
+    track: string,
+    versionCodes: number[],
+    versionCodeFilterType: string,
+    versionCodeFilter: string | number[],
+    userFraction: number,
+    updatePriority: number,
+    releaseNotes?: pub3.Schema$LocalizedText[],
+    releaseName?: string
```

Comment:
Could it make sense to combine all these inputs into a single object?

Reply:
Take a look at https://github.com/microsoft/google-play-vsts-extension/pull/292/commits/e138763b71337ce4ef23d552d164b668d5726e81
I was hesitant to do that since I don't like the look of the resulting typescript code. I could leave it or revert it. What do you think is the better option?

Reply:
Reworked in https://github.com/microsoft/google-play-vsts-extension/pull/292/commits/79b751997799937003df25ca7612984ef23f11db